ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/activities/ExpenseItemActivity.java

```diff
@@ -367,9 +378,22 @@ public void onClick(DialogInterface dialog, int which) {
 			item.setCurrency(currency);
 			item.setDescription(description);
 			item.setReceiptPhoto(bm);
+			if (!expenseItemGeolocation.equals(new Coordinate())) {
```

Comment:
This code should also be checking against null, similar to the change made to AddDestinationActivity.